### PR TITLE
Apply some Pylint fixes, Part 1: Clean up parser

### PR DESCRIPTION
**Pull request by EmilyBourne**
The first of a few PRs which merge the changes from #1448 . This PR cleans up the parser/ folder by fixing missing fstrings . As a result docstrings are also improved.

### CODE REVIEW

File: pyccel/parser/scope.py [hunk 1]
```diff
@@ -308,21 +308,17 @@ def create_new_loop_scope(self):
...
-        python_scope : bool
-                If true then we assume that python scoping applies.
-                In this case variables declared in loops exist beyond
-                the end of the loop. Otherwise variables may be local
-                to loops
-                Default : True
+        var : Variable
+            The variable to be inserted into the current scope.
+        name : str, default=var.name
+            The name of the variable in the python code.
```
Comment (yguclu) [inline pyccel/parser/scope.py:321]: Maybe capitalize Python?

File: pyccel/parser/scope.py [hunk 2]
```diff
@@ -749,15 +765,28 @@ def add_son(self, name, son):
         self._sons_scopes[name] = son
 
     def get_python_name(self, name):
-        """ Get the name used in the original python code from the
-        name used by the variable
+        """
+        Get the name used in the original python code.
+
+        Get the name used in the original python code from the name used
+        by the variable that was created in the parser.
```
Comment (yguclu) [inline pyccel/parser/scope.py:772]: Maybe capitalize Python?

File: pyccel/parser/base.py [hunk 3]
```diff
@@ -478,13 +488,16 @@ def dump(self, filename=None):
...
-          filename: str
-            output file name. if not given `name.pyccel` will be used and placed
-            in the Pyccel directory ($HOME/.pyccel)
+        Load the current ast using Pickle.
+
+        Parameters
+        ----------
+        filename : str
+            Output file name. if not given `name.pyccel` will be used and placed
+            in the Pyccel directory ($HOME/.pyccel).
```
Comment (EmilyBourne) [inline pyccel/parser/base.py:500]:
```suggestion
            The name of the pickled file. if not given `name.pyccel` will be used.
```

File: pyccel/parser/base.py [hunk 4]
```diff
@@ -478,13 +488,16 @@ def dump(self, filename=None):
...
-          Parameters
-          ----------
-          filename: str
-            output file name. if not given `name.pyccel` will be used and placed
-            in the Pyccel directory ($HOME/.pyccel)
+        Load the current ast using Pickle.
+
+        Parameters
+        ----------
+        filename : str
```
Comment (EmilyBourne) [inline pyccel/parser/base.py:498]:
```suggestion
        filename : str, optional
```

File: pyccel/parser/scope.py (hunk 1 above)
Comment (EmilyBourne) [inline pyccel/parser/scope.py:321]:
```suggestion
            The name of the variable in the Python code.
```

File: pyccel/parser/scope.py [hunk 5]
```diff
@@ -749,15 +765,28 @@ def add_son(self, name, son):
         self._sons_scopes[name] = son
 
     def get_python_name(self, name):
-        """ Get the name used in the original python code from the
-        name used by the variable
+        """
+        Get the name used in the original python code.
+
+        Get the name used in the original python code from the name used
```
Comment (EmilyBourne) [inline pyccel/parser/scope.py:771]:
```suggestion
        Get the name used in the original Python code.

        Get the name used in the original Python code from the name used
```

File: pyccel/parser/scope.py (hunk 1 above)
Comment (EmilyBourne) [reply pyccel/parser/scope.py]: Done c2b07b36dbc71476b2c175d044207c30e1db7340

**Review by yguclu — APPROVED** [review]

**Pull request merged**